crates/flipperzero/src/dialogs/mod.rs

```diff
@@ -164,6 +202,84 @@ impl DialogMessageButton {
     }
 }
 
+impl<'a> DialogFileBrowserOptions<'a> {
+    /// Creates a new dialog file browser options and initializes to default values.
+    pub fn new(extension: Option<&'a CStr>, icon: Option<&'a sys::Icon>) -> Self {
+        let mut dialog_file_browser_options = Self {
+            data: unsafe { mem::zeroed() },
```

Comment:
I was following the pattern found in C code of creating an uninitialized struct: `DialogsFileBrowserOptions = {};` and passing it to `dialog_file_browser_set_basic_options()` to initialize.

I can initialize all the struct values how `dialog_file_browser_set_basic_options()` does in Rust and avoid the call into C. The options can then be customized with the setter methods. I think I'll remove the `Default` trait implementation because `new()` will be the default.

Reply:
Check out the current change. By the way, in the Furi C code, some pointer values are allowed to be NULL. For example, the `base_path`, `icon, and `item_loader_context` can all be NULL, these are checked in the C code. The `extension` member has to be a valid string. When calling `sys::dialog_file_browser_show()`, you can also pass NULL for the file browser options.

Reply:
My apologize for a bit chaotic review :heart: